pycernan/avro/client.py

```diff
@@ -1,47 +1,126 @@
 """
     Base Avro client from which all other clients derive.
 """
+import contextlib
 import socket
 
 import pycernan.avro.config
 
 from abc import ABCMeta, abstractmethod
+from queue import Queue, Empty
 
-from pycernan.avro.exceptions import EmptyBatchException
+from pycernan.avro.exceptions import EmptyBatchException, EmptyPoolException
 from pycernan.avro.serde import serialize
 
 
+_DefunctConnection = object()
+
+
+class TCPConnectionPool(object):
+    """
+    Fork friendly TCP connection pool.
+
+    Adapted from: https://github.com/gevent/gevent/blob/master/examples/psycopg2_pool.py
+    """
+    def __init__(self, host, port, maxsize, connect_timeout, read_timeout):
+        if maxsize <= 0:
+            raise ValueError("maxsize must be > 0")
+
+        self.host = host
+        self.port = port
+        self.size = 0
```

Comment:
You don't need this anymore right?

Reply:
Yep.  This needs to go.  Thanks for catching this.